shared/src/ledger/storage/write_log.rs

```diff
@@ -7,6 +7,7 @@ use thiserror::Error;
 
```

Comment:
regarding the changes in this module:
- can there only be up to 1 IBC event per tx or should be it possible to have multiple events? Maybe it could be a vec
- I think it would be better to separate the IBC event from `StorageModification` - we could add a new field in the `WriteLog` instead (the key is constant, so we don't need to use it at all and I don't think we should `#[allow(clippy::large_enum_variant)]` on `StorageModification` as this is on a hot path)

Reply:
> can there only be up to 1 IBC event per tx or should be it possible to have multiple events?
only 1 IBC event per tx

> I think it would be better to separate the IBC event from `StorageModification`
That makes sense. Thanks!